src/rt-app.c

```diff
@@ -976,6 +976,23 @@ static void set_thread_priority(thread_data_t *data, sched_data_t *sched_data)
 		sa_params.sched_flags = SCHED_FLAG_KEEP_ALL;
 		tid = gettid();
 
+		/*
+		 * Currently we cannot use the syscall to set a clamp
+		 * value without specifing a valid policy and priotiry.
+		 * Read the current values and use them for the following
+		 * syscall.
+		 * TODO: fix __sched_setscheduler() to get rid of this!
+		 */
+		ret = sched_getattr(tid, &sa_params,
```

Comment:
I think sa_params is only present if DL is enabled - I didn't see anything moving the declaration outside the DLSCHED guards.

Reply:
Hmm I thought @derkling was going to remove the conditional compilation of deadline and always compile it in